### PR TITLE
Support single execution python function in Pulsar

### PR DESCRIPTION
**Pull request by odmarkj**
This modifies pulsar to allow a function to run once, then shut itself down through the REST API.

This was built to support PandioML.

Still need to figure out how to authenticate the request to the REST API to shut a function down.

The pulsar client is able to authenticate in the python_instance_main.py file, so I need to figure out how to leverage this same authentication if possible in an HTTP POST request for Pandio's platform.

### CODE REVIEW

**Comment by odmarkj** [comment]
@deonvdv @deltasquare4 Hoping to get your opinion on the best way to test this on the Pandio platform. Might need to ask @oginskis for help in figuring out how to authenticate to the Pandio Pulsar ADMIN REST API from inside of these python files.

Basically need to figure out how to extract the JWT, and/or use the same authentication that the pulsar.Client uses.

**Review by deltasquare4 — COMMENTED** [review]
@odmarkj Looks good to me. I can embed it in a build. Where would you like me to deploy it?

**Comment by odmarkj** [comment]
Before we put it into a build, I need help figuring out how to authenticate this request:

https://github.com/pandio-com/pulsar/blob/bb25fff74bd161970b460d3dece8cae5a3867f6b/pulsar-functions/instance/src/main/python/python_instance.py#L140

I already do this inside of PandioML by setting the JWT in the header, but I don't want to embed a users JWT in the function for security reasons.

The pulsar client that a function uses has authentication here: https://github.com/pandio-com/pulsar/blob/bb25fff74bd161970b460d3dece8cae5a3867f6b/pulsar-functions/instance/src/main/python/python_instance_main.py#L149-L161

I need help figuring out how to use this to extract information to authorize a request to the ADMIN API.

Or, if you can build this into an image, and then give me access to SSH into the pod that has the function running inside of it, I might be able to debug live. I can't replicate the authentication with Pulsar locally, that would be ideal.

Basically need to get the service account information inside of a function for the user that uploads the function.

**Comment by oginskis** [comment]
Function talks to Pulsar on behalf of a user who created the function. JWT token of a user is mounted to `/etc/auth/token` location within function pod. You can read it from there. @odmarkj

**Comment by odmarkj** [comment]
Excellent @oginskis !

Is the JWT the only thing in that file or is it JSON that needs to be parsed to get the JWT?

**Comment by oginskis** [comment]
The file contains JWT only

**Comment by odmarkj** [comment]
Alright, added the token. @deltasquare4 I would like to get this into production asap, so if we have to put it into 2.6.0 that is fine. Or if 2.7.1 will be ready for production in the next week or two, that is fine.

**Comment by deltasquare4** [comment]
Merged manually to `pandio-2.7` branch.
